File: src-old/test/java/io/tripled/social/client/infrastructure/TestDateTimeProvider.java
package io.tripled.social.client.infrastructure;

import io.tripled.social.client.domain.DateTimeProvider;

public class TestDateTimeProvider implements DateTimeProvider {

  private Clock clock;

  public TestDateTimeProvider(Clock clock) {
    this.clock = clock;
  }

  @Override
  public LocalDateTime now() {
    return LocalDateTime.now(clock);
  }

  public void fixate(Instant instant) {
    clock = Clock.fixed(instant, ZoneId.systemDefault());
  }

  public void fixateWithOffset(Duration duration) {
    clock = Clock.offset(clock, duration);
  }
}

File: src/test/kotlin/io/tripled/social/client/infrastructure/time/TestDateTimeProvider.kt
package io.tripled.social.client.infrastructure.time

import io.tripled.social.client.domain.DateTimeProvider
import java.time.Clock
import java.time.Duration
import java.time.Instant
import java.time.LocalDateTime
import java.time.ZoneId

class TestDateTimeProvider(private var clock: Clock) : DateTimeProvider {
    override fun now(): LocalDateTime {
        return LocalDateTime.now(clock)
    }

    fun fixate(instant: Instant) {
        clock = Clock.fixed(instant, ZoneId.systemDefault())
    }

    fun fixateWithOffset(duration: Duration) {
        clock = Clock.offset(clock, duration)
    }
}